tools/autograd/gen_variable_type.py

```diff
@@ -131,7 +136,33 @@
 """)
 
 RECORD_ATTRIBUTE = CodeTemplate("""\
-        setattr(trace_info.n, jit::attr::${name}, ${name});""")
+setattr(trace_info.n, jit::attr::${name}, ${name});""")
+
+RECORD_POSITIONAL_ATTRIBUTE = CodeTemplate("""\
+setposattr(trace_info.n, ${i}, "${name}", ${name});""")
+
+POSITIONAL_ATTR_NYI = """\
+throw std::runtime_error("Can't have size-dependent arguments to functions that "
+                         "take variable number of tensor arguments");
+"""
+
+
+def should_trace(declaration):
+    # Operations involving Generator, Storage, Type are not traceable
+    # at the moment
+    if any(arg['simple_type'] in {'Generator', 'Storage', 'ScalarType', 'Type'} for arg in declaration['arguments']):
```

Comment:
(not related to this diff per se)
hm, what exactly does it mean? is it generic jit tracing or only onnx export? We'd need to figure out a way to handle opaque objects in tracing at some point too.

Reply:
ONNX export goes through JIT tracing, so it's both. FWIW this only relaxes the existing constraints, and isolates them in this single function, because it's not needed at multiple stages of codegen.